sdk/communication/AzureCommunicationChat/Source/ChatClient.swift

```diff
@@ -66,6 +68,20 @@ public class ChatClient {
         )
 
         self.service = client.chat
+
+        // Initialize the signalling client with the users access token
+        var token: String?
+        var tokenError: Error?
+        credential.token { accessToken, error in
+            token = accessToken?.token
+            tokenError = error
+        }
+
+        guard let skypeToken = token else {
+            throw AzureError.client("Failed to access token credential", tokenError)
+        }
```

Comment:
I was confused when I first saw this too but after running it myself it's not async @JoshuaLai does usage look correct to you here?

Reply:
It is async. The closure returns the token so this guard should happen in the closure as Travis mentioned

Reply:
Thanks, I'll change this! Weird to me how did this work before though? 

We have been using a test app @glorialimicrosoft made and the logic is always like this, I never had a failure running it though:  https://github.com/knvsl/azure-sdk-for-ios/blob/4687286c440ac767247f22471c6c7be3d1553e3e/sdk/communication/AzureCommunicationChat/Source/ChatClient.swift#L80

The original too:
https://github.com/Azure/azure-sdk-for-ios/blob/153a6b58de2807f43ce57c5375107a46feab0186/sdk/communication/AzureCommunicationChat/Source/ChatClient.swift#L240

Reply:
Hmmm it could be returning to you faster if there was a cached instance of the token that hasn't expired yet.  But it is much safer to rely on the token in the closure

Reply:
It seems then like all of the following logic should be moved into the closure.

Reply:
I went with moving it into startRealtimeNotifications() to avoid the async call in init()